cmd/sealer/cmd/cluster/join.go

```diff
@@ -87,6 +90,13 @@ func NewJoinCmd() *cobra.Command {
 				return err
 			}
 
+			defer func() {
+				//if there exits an error,rollback the ClusterFile to the default file
+				if err = cf.RollBackClusterFile(err); err != nil {
```

Comment:
There is no need to pass error to the function.
This function does not have to be aware of error.
if err == nil {
  return
}

RollBackClusterFile()
...